### PR TITLE
[PhpUnitBridge] add PolyfillTestCaseTrait::expectExceptionMessageMatches to provide FC with recent phpunit versions

### PR DESCRIPTION
**Pull request by soyuka**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Tickets       | na
| License       | MIT
| Doc PR        | na

expectExceptionMessageRegExp is deprecated coming phpunit 8.5.3  see https://github.com/sebastianbergmann/phpunit/issues/4133

Not sure if I need to add something else lmk.

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
I suppose the codebase should be updated accordingly?

**Comment by soyuka** [comment]
you mean change expectExceptionMessageRegExp to expectExceptionMessageMatches?

**Comment by dunglas** [comment]
@soyuka yes, in the whole Symfony code base.

File: src/Symfony/Component/Yaml/Tests/ParserTest.php [hunk 1]
```diff
@@ -2312,7 +2312,7 @@ public function testMultiLineComment()
 parameters:
     abc
 
-# Comment 
+# Comment
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Yaml/Tests/ParserTest.php:2315]: Should be reverted

File: src/Symfony/Component/Process/Tests/ProcessTest.php [hunk 2]
```diff
@@ -1439,7 +1439,7 @@ public function testRawCommandLine()
 (
     [0] => -
     [1] => a
-    [2] => 
+    [2] =>
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Process/Tests/ProcessTest.php:1442]: Should be reverted

File: composer.json [hunk 3]
```diff
@@ -156,6 +156,10 @@
         {
             "type": "path",
             "url": "src/Symfony/Contracts"
+        },
+        {
+            "type": "path",
+            "url": "src/Symfony/Bridge/PhpUnit"
```
Comment (soyuka) [inline composer.json:162]: Totally unsure about this, I don't see how we'll get unit tests to pass without the update on the bridge. Maybe that this deserves a second pull request?

**Comment by nicolas-grekas** [comment]
Thank you @soyuka.

**Comment by nicolas-grekas** [comment]
@soyuka would you mind contributing a rule to php-cs-fixer so that this is enforced automatically in the future please?

**Pull request merged**